### DISCUSSION TITLE
One Broker, multi port(tcp) is possible?

### DISCUSSION THREAD

**Discussion by JeongWooRam** [emqx/emqx#13224] [Q&A]
I added listeners for tcp port 1884 and tcp port 1885 in addition to 1883 in the dashboard.
But when I publish the message to 1884 or 1885, the message arrives on all tcp ports (same topic).

If I publish a message to tcp port 0.0.0.0:1885(external), I want it to be received only at 127.0.0.1:1885(internal). (No received 1883 or 1884)

Is it possible to change the settings?

**Comment by qzhuyan** [comment]
I don't think you could simply isolate traffic among the listeners.

**Comment by zmstone** [comment]
@JeongWooRam this is possible with `zone`, `ACL` control (emqx 5.7 or later). 

Put two listeners in the same zone, so the clients connected to the same zone will be attributed with the same value.
For how to configure zone and client attributes, you can refer to this thread: https://github.com/emqx/emqx/issues/13159#issuecomment-2139500581

An alternative (as well as to make it easier to configure ACL rules), you can also consider adding the same `mountpoint` for the listeners intended to work in paired-up mode.
see: https://docs.emqx.com/en/enterprise/v5.7.0/hocon/#V-listeners-S-listeners-tcp-S-mqtt_tcp_listener-mountpoint
